framework/Models/ROM.py

```diff
@@ -267,7 +1134,93 @@ def __init__(self,runInfoDict):
     self.amITrained                = False      # boolean flag, is the ROM trained?
     self.supervisedEngine          = None       # dict of ROM instances (== number of targets => keys are the targets)
     self.printTag = 'ROM MODEL'
+    # list of Keras Neural Network Core layers
```

Comment:
Here is for the input checks. We currently do not do input checks in subclasses.

Reply:
This is still way too many lines for the base class - the number of added input spec lines here exceeds the total original length of the model ROM code, twice over. 

I talked to the CCB chair designee, and he says we can merge this now, but we need to set as high priority the restructuring of the ROMs so that these input specs can be split out into their individual members.